### ISSUE TITLE
Improve Typing in ConstitutiveLaw abstract class

### ISSUE THREAD

**Issue by DanielGMorenaFhecor** [fib-international/structuralcodes#111]
I have realized that the `get_tangent(self, eps: float)` method in the `ConstitutiveLaw` abstract class has a `float` as a return type but all implementations of this class have the following signature:

```python
def get_tangent(self, eps: ArrayLike) -> ArrayLike
    pass
```

It would be nice to change it for clarity.

**Comment by mortenengen** [comment]
Thanks for pointing it out @DanielGMorenaFhecor. I have suggested a fix in #112 both for `get_stress` and `get_tangent`.

**Comment by talledodiego** [comment]
You are right!
my bad I did not go back to the abstract class.
Thanks @mortenengen for the fix.
